src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -19,35 +20,84 @@
+
+    initialBallSensor = new DigitalInput(Constants.initialBallSensorChannel);
+    middleBallSensor = new DigitalInput(Constants.middleBallSensorChannel);
+    finalBallSensor = new DigitalInput(Constants.finalBallSensorChannel); 
 
-    intakeMotorControllerTwo.getSparkMax().follow(intakeMotorControllerTwo.getSparkMax());
+    DigitalInput[] sensorArray = {initialBallSensor, middleBallSensor, finalBallSensor};
+
+    // Remove invert=true parameter if wheels aren't running correctly
+    //CDSWheelControllerOne.getSparkMax().follow(intakeMotorControllerOne.getSparkMax());
+    //CDSWheelControllerTwo.getSparkMax().follow(intakeMotorControllerOne.getSparkMax(), true);
+    intakeMotorControllerTwo.getSparkMax().follow(intakeMotorControllerOne.getSparkMax());
   }
 
-  public void IntakeSwitch(boolean on){    
-    if (on){
-      double intakeSmartSpeed = SmartDashboard.getNumber("Belt Speed", Constants.CDSBeltSpeed);
-      
-      intakeMotorControllerOne.getSparkMax().set(intakeSmartSpeed);
+  public void toggleIntake(boolean reverse) {
+    if (reverse) {
+      intakeMotorControllerOne.getSparkMax().set(-Constants.intakeMotorSpeed);
+      SmartDashboard.putString("Intake Motor Direction", "Reverse");
+      SmartDashboard.putNumber("Intake Motor Speed", -Constants.intakeMotorSpeed);
+    } else {
+      intakeMotorControllerOne.getSparkMax().set(Constants.intakeMotorSpeed);
+      SmartDashboard.putString("Intake Motor Direction", "Forward");
       SmartDashboard.putNumber("Intake Motor Speed", Constants.intakeMotorSpeed);
+    }
+  }
+
+  public void stopIntake() {
+    intakeMotorControllerOne.getSparkMax().set(0.0);
+    SmartDashboard.putNumber("Intake Motor Speed", 0.0);
+  }
+
+  public boolean getDirection() {
+    // true = inverted, false = forward
+    if (intakeMotorControllerOne.getSparkMax().get() > 0) {
+      return false;
     } else {
-      intakeMotorControllerOne.getSparkMax().set(0);
-      SmartDashboard.putNumber("Intake Motor Speed", 0);
+    return true;
     }
   }
-  
-  public void ForwardIntake(){
-    intakeMotorControllerOne.getSparkMax().setInverted(false);
-    SmartDashboard.putString("Intake Motor Direction", "Forward");
+
+  public boolean[] getBeamBreakStatus() {
+    boolean initialStatus = initialBallSensor.get();
+    boolean middleStatus = middleBallSensor.get();
+    boolean finalStatus = finalBallSensor.get();
+    boolean[] beamBreakArray = {initialStatus, middleStatus, finalStatus};
+    return beamBreakArray;
   }
 
-  public void ReverseIntake(){
-    intakeMotorControllerOne.getSparkMax().setInverted(true);
-    SmartDashboard.putString("Intake Motor Direction", "Reverse");
+  public int getBallCount() {
+    boolean[] beamBreakStatuses = this.getBeamBreakStatus();
+    if (!beamBreakStatuses[0]) {
+      if (this.getDirection()){
+        ballCount--;
+      } else {
+        ballCount++;
+      }
+    }
+    return ballCount;
   }
```

Comment:
A few things here:
1. It seems like the ball count should just count the beam breaks rather than try to detect which way the motion is. At least add some comments to explain the intent. 
2. If the intake is not moving, and we have 1 ball stored, ballCount will keep incrementing and cause us to spit it out. Set ballCount to 0 before checking all the statuses.
3. I think we talked about only using the first beam break to testing right now, but add a TODO comment to update it later.